lib/backend/k8s/resources/networkpolicy.go

```diff
@@ -130,26 +166,48 @@ func (c *networkPolicyClient) Get(ctx context.Context, key model.Key, revision s
 		return nil, errors.New("Missing policy namespace")
 	}
 
+	// The revision, if supplied, will be a combination of CRD and k8s-backed revisions.  Extract
+	// the k8s rev and use the correct version depending on whether we are querying the CRD or the
+	// k8s NetworkPolicy.
+	crdRev, k8sRev, err := splitRevision(revision)
+	if err != nil {
+		return nil, err
+	}
+
 	// Check to see if this is backed by a NetworkPolicy.
 	if strings.HasPrefix(k.Name, conversion.K8sNetworkPolicyNamePrefix) {
 		// Backed by a NetworkPolicy - extract the name.
 		policyName := strings.TrimPrefix(k.Name, conversion.K8sNetworkPolicyNamePrefix)
 
 		// Get the NetworkPolicy from the API and convert it.
 		networkPolicy := extensions.NetworkPolicy{}
-		err := c.clientSet.Extensions().RESTClient().
+		err = c.clientSet.Extensions().RESTClient().
 			Get().
 			Resource("networkpolicies").
 			Namespace(k.Namespace).
 			Name(policyName).
-			VersionedParams(&metav1.GetOptions{ResourceVersion: revision}, scheme.ParameterCodec).
+			VersionedParams(&metav1.GetOptions{ResourceVersion: k8sRev}, scheme.ParameterCodec).
 			Do().Into(&networkPolicy)
 		if err != nil {
 			return nil, K8sErrorToCalico(err, k)
 		}
-		return c.converter.K8sNetworkPolicyToCalico(&networkPolicy)
+		kvp, err := c.converter.K8sNetworkPolicyToCalico(&networkPolicy)
+
+		// Convert the revision back to the combined CRD/k8s revision - the CRD rev will be empty.
+		if kvp != nil {
+			kvp.Revision = joinRevisions("", kvp.Revision)
+		}
+		return kvp, err
 	} else {
-		return c.crdClient.Get(ctx, k, revision)
+		// The revision, if supplied, will be a combination of CRD and k8s-backed revisions.  Extract
```

Comment:
comment seems wrong, we're not extracting it here.

Reply:
Ah! I originally had the extraction code in both branches and moved it out - looks like i didn't move the comment - my bad.  Removing!